libs/utils/gtest/src/ThreadsTestSuite.cc

```diff
@@ -278,6 +284,30 @@ TEST_F(ThreadsTestSuite, CondBroadcastTest) {
     free(param);
 }
 
+TEST_F(ThreadsTestSuite, CondTimedWaitTest) {
+    celix_thread_mutex_t mutex;
+    celix_thread_cond_t cond;
+
+    auto status = celixThreadMutex_create(&mutex, nullptr);
+    ASSERT_EQ(status, CELIX_SUCCESS);
+    status = celixThreadCondition_init(&cond, nullptr);
+    ASSERT_EQ(status, CELIX_SUCCESS);
+
+    //Test with nullptr abstime
+    status = celixThreadCondition_waitUntil(&cond, &mutex, nullptr);
+    ASSERT_EQ(status, CELIX_ILLEGAL_ARGUMENT);
+
+    //Test with valid abstime
+    auto start = celixThreadCondition_getTime();
+    auto targetEnd = celixThreadCondition_getDelayedTime(0.01);
+    celixThreadMutex_lock(&mutex);
+    status = celixThreadCondition_waitUntil(&cond, &mutex, &targetEnd);
+    ASSERT_EQ(status, ETIMEDOUT);
+    celixThreadMutex_unlock(&mutex);
+    auto end = celixThreadCondition_getTime();
+    EXPECT_NEAR(celix_difftime(&end, &start), 0.001, ALLOWED_ERROR_MARGIN_IN_MS);
```

Comment:
Does it means that  20 *seconds* is a permissible error?

Reply:
Quick fix https://github.com/apache/celix/pull/583/commits/b812c6b21fd0c87bac4f840a529fa3ea3e52fd31